mailexporter.go

```diff
@@ -401,29 +415,39 @@ func main() {
 	if err != nil {
 		promlog.Fatal(err)
 	}
+	defer fileClose(f)
 
 	err = parseConfig(f)
-	f.Close()
 	if err != nil {
 		promlog.Fatal(err)
 	}
 
 	// initialize Metrics that will be used seldom so that they actually get exported with a metric
 	for _, c := range globalconf.Servers {
-		lateMails.GetMetricWithLabelValues(c.Name)
-		mailSendFails.GetMetricWithLabelValues(c.Name)
+
+		_, err1 := lateMails.GetMetricWithLabelValues(c.Name)
```

Comment:
Instead of handling the error here, we should actually use the function `WithLabelValues` instead of `GetMetricWithLabelValues` (function signature stays the same).

This will not return an error but panic in case of an error. If that happens here it will happen anyways later, because `WithLabelValues` is used throughout the code. So I think triggering the then unavoidable ASAP would be the better approach to address this.